model/src/lib.rs

```diff
@@ -61,44 +55,31 @@ impl Signature {
         }
     }
 
-    fn signing_digest_typed_data(
+    pub fn validate(
+        &self,
         domain_separator: &DomainSeparator,
-        digest: &[u8; 32],
-    ) -> [u8; 32] {
-        let mut hash_data = [0u8; 66];
-        hash_data[0..2].copy_from_slice(&[0x19, 0x01]);
-        hash_data[2..34].copy_from_slice(&domain_separator.0);
-        hash_data[34..66].copy_from_slice(digest);
-        signing::keccak256(&hash_data)
+        struct_hash: &[u8; 32],
+    ) -> Option<H160> {
+        let message = hashed_eip712_message(domain_separator, struct_hash);
+        let recovery = Recovery::new(message, self.v as u64, self.r, self.s);
+        let (signature, recovery_id) = recovery.as_signature()?;
+        signing::recover(&message, &signature, recovery_id).ok()
     }
 
-    pub fn signing_digest_message(
+    pub fn sign(
```

Comment:
Is there still a unit test that check we can sign and later validate the signers address?

Reply:
Yes, `order_builder_signature_recovery` in `order.rs`.